consensus/hotstuff/votecollector/common.go

```diff
@@ -43,3 +45,28 @@ func EnsureVoteForBlock(vote *model.Vote, block *model.Block) error {
 	}
 	return nil
 }
+
+// ConcurrentIdentifierSet implements a simple set for tracking unique entries by identifier.
+// Concurrency safe.
+type ConcurrentIdentifierSet struct {
```

Comment:
In other words, we specifically use this set in scenarios where we specifically desire the append-only guarantees. But that's not obvious from the implementation. Assume somebody decided that this was a nice concurrent set, that also needed other set operations like element removal. Thereby, most places currently using this set would not out of the box benefit from the "append-only" guarantees of the set. 

To preempt that, I would **suggest** we **rename** this to **`AppendOnlyIdentifierSet`**

Reply:
I thought that in future we could expand this with more functions when it's needed, but we can also go with the renaming route.